libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -320,6 +320,58 @@ public new BotFrameworkAdapter Use(IMiddleware middleware)
             await connectorClient.Conversations.DeleteActivityAsync(reference.Conversation.Id, reference.ActivityId);
         }
 
+        /// <summary>
+        /// Attempts to retrieve the token for a user that's in a login flow.
+        /// </summary>
+        /// <param name="context">Context for the current turn of conversation with the user.</param>
+        /// <param name="connectionName">Name of the auth connection to use.</param>
+        /// <param name="magicCode">(Optional) Optional user entered code to validate.</param>
+        /// <returns></returns>
+        public async Task<TokenResponse> GetUserToken(ITurnContext context, string connectionName, string magicCode)
+        {
+            if (context == null)
```

Comment:
There's a BotAssert.ContextNotNull() method that will do this. Please use that.

Reply:
done